skimage/filters/tests/test_thresholding.py

```diff
@@ -501,6 +502,25 @@ def test_niblack_sauvola_pathological_image():
     assert not np.any(np.isnan(threshold_niblack(src_img)))
 
 
+def test_bimodal_multiotsu_hist():
+    image = data.camera()
+    thr_otsu = threshold_otsu(image)
+    thr_multi = threshold_multiotsu(image, classes=2)
+    assert thr_otsu == thr_multi
+
+
+def test_check_multiotsu_results():
```

Comment:
this test takes for ever on my laptop. Is it entirely necessary?

Reply:
Is the fact that this test is taking so long speaking that the implementation might be inefficient? I really don't know, but it might be something to ponder about.

Reply:
I'm not sure if it is a timeout, but I definitely thought my computer was frozen while running these tests. If you use pytest parametrize like the test below, it will give the test suite 3 outputs which will give Travis a sense of progress and might not time out.

Reply:
```patch
-def test_check_multiotsu_results():
+@pytest.mark.parametrize("classes", range(3, 6))
+def test_check_multiotsu_results(classes):
     image = 0.25 * np.array([[0, 1, 2, 3, 4],
                              [0, 1, 2, 3, 4],
                              [0, 1, 2, 3, 4],
                              [0, 1, 2, 3, 4]])
 
-    for idx in range(3, 6):
-        thr_multi = threshold_multiotsu(image,
-                                        classes=idx)
-        assert len(thr_multi) == idx-1
+    thr_multi = threshold_multiotsu(image,
+                                    classes=classes)
+    assert len(thr_multi) == classes - 1
```